script_test.go

```diff
@@ -1768,6 +1769,57 @@ func TestWithStdout_SetsSpecifiedWriterAsStdout(t *testing.T) {
 	}
 }
 
+func TestWithEnv_SetEmptyEnvironment(t *testing.T) {
+	t.Parallel()
+	buf := new(bytes.Buffer)
+	env := []string{}
+
+	_, err := script.NewPipe().WithStdout(buf).WithEnv(env).Exec("printenv").Stdout()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(buf.String()) != 0 {
+		got := strings.Split(strings.TrimRight(buf.String(), "\n"), "\n")
+		t.Errorf("expected 0 environment variables, got %d %v", len(got), got)
+	}
+}
+
+func TestWithEnv_SetMultipleEnvVars(t *testing.T) {
+	t.Parallel()
+	buf := new(bytes.Buffer)
+	env := []string{"ENV1=test1", "ENV2=test2"}
+
+	_, err := script.NewPipe().WithStdout(buf).WithEnv(env).Exec("printenv").Stdout()
```

Comment:
It seems a shame to add an implicit dependency on the `printenv` command, doesn't it? Could we use the shell instead?

Reply:
@bitfield would you please be able to elaborate on this for me? i'm not sure what it means to use the shell instead

Reply:
Well, you'll notice that `script` itself doesn't rely on _any_ external commands being present—which is a good thing, because we basically can't rely on that! And the tests don't use many external commands either: currently just `go` (which seems at least reasonable) and `echo` (which exists most places, is usually a shell builtin, and even works on Windows, though with slightly different syntax).

If we could stick to those, without adding a new dependency that we don't absolutely need, that would be a nice bonus.

Reply:
@bitfield there is the `env` command but i'm not sure that's too different from using `printenv`. the other option would be to use `echo $ENV1` to capture the value of a single environment variable but i don't believe there's a way to print all the variables using `echo`. let me know what you think!

Reply:
If we could avoid adding a new external dependency, that would be great!

Reply:
@bitfield using `echo` doesn't actually work since the variables aren't exported. I'm not sure if there's another way to print the variables we want without using `env` or `printenv`.

Reply:
Yes, this is a bit of a subtle one. If you did something like this:

```go
Exec("echo ENV1=$ENV1")
```

that wouldn't work, because `$`-expansion is done by the _shell_, and we're not executing a shell! We're executing the `echo` command directly, which doesn't interpolate variables.

On the other hand, if we exec a shell instead:

```go
Exec("sh -c 'echo ENV1=$ENV1'")
```

this works as expected, because the shell expands `$ENV1` before passing it to `echo`.

We do use `sh -c` in a couple of existing tests, so there's no problem with depending on it here.

Reply:
That's really cool! Thanks for the explanation @bitfield